kedro/extras/datasets/snowflake/snowflake_dataset.py

```
@@ -0,0 +1,252 @@
+"""``SnowflakeTableDataSet`` to load and save data to a snowflake backend."""
+
+import copy
+import re
+from typing import Any, Dict, Optional
+
+import pandas as pd
+from snowflake.snowpark import Session
+
+from kedro.io.core import AbstractDataSet, DataSetError
+
+KNOWN_PIP_INSTALL = {
+    "snowflake.snowpark": "snowflake.snowpark",
+}
+
+DRIVER_ERROR_MESSAGE = """
+A module/driver is missing when connecting to Snowflake
+\n\n
+"""
+
+
+def _find_known_drivers(module_import_error: ImportError) -> Optional[str]:
+    """Looks up known keywords in a ``ModuleNotFoundError`` so that it can
+    provide better guideline for the user.
+
+    Args:
+        module_import_error: Error raised while connecting to a SQL server.
+
+    Returns:
+        Instructions for installing missing driver. An empty string is
+        returned in case error is related to an unknown driver.
+
+    """
+
+    # module errors contain string "No module name 'module_name'"
+    # we are trying to extract module_name surrounded by quotes here
+    res = re.findall(r"'(.*?)'", str(module_import_error.args[0]).lower())
+
+    # in case module import error does not match our expected pattern
+    # we have no recommendation
+    if not res:
+        return None
```

Do we want to raise an error here?